src/complex/DataStructure/AbstractDataStore.hpp

```diff
@@ -27,6 +27,9 @@ template <typename T>
 class AbstractDataStore : public IDataStore
 {
 public:
+  static constexpr const char k_TupleShape[] = "TupleShape";
+  static constexpr const char k_ComponentShape[] = "ComponentShape";
```

Comment:
```suggestion
  static inline constexpr const char k_TupleShape[] = "TupleShape";
  static inline constexpr const char k_ComponentShape[] = "ComponentShape";
```
Maybe move these into `IDataStore`. That way they can be accessed without having to specify a template type.